docs/extensions/oauth2-flow.md

```diff
@@ -0,0 +1,110 @@
+---
+title: OAuth 2.0 flow
+description: Docker extension OAuth 2.0 flow
+keywords: Docker, extensions, sdk, OAuth 2.0
+---
+
+You may need users to authenticate from your Docker Extension using OAuth 2.0 via the web browser, and return them back into the Docker Extension.
+
+> ** Note **
+>
+> This page assumes that you already have an identity provider (IdP) which handles the authentication process and returns an access token, like Google, Azure AD or Okta, among many others.
```

Comment:
@mikeparker LMK if this paragraph is enough to clarify users must provide their own OAuth endpoints 😄